MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -306,6 +307,30 @@ open class RouteLegProgress: NSObject {
         assert(!nearby.isEmpty, "Step must have coordinates")
         return nearby
     }
+    
+    typealias StepIndexDistance = (index: Int, distance: CLLocationDistance)
+    
+    func closestStep(to coordinate: CLLocationCoordinate2D) -> StepIndexDistance? {
```

Comment:
I think the performance of this method could be improved substantially if we add a pruning radius argument to this method. That way, we can filter off any step that's outside of our "radius" before enumerating through the array.

L313:
```swift
func closestStep(to coordinate: CLLocationCoordinate2D, within radius: CLLocationDistance = .greatestFiniteMagnitude) -> StepIndexDistance?
```

L315: 
```swift
let remainingSteps = leg.steps.suffix(from: stepIndex)
let filteredSteps = remainingSteps.filter({ $0.distance <= radius})
```

then in L317, enumerate on `filteredSteps`. 

Considering that `Array.suffix` returns an `ArraySlice`, you _shouldn't_ need to manually keep track of array indices by doing this, but don't quote me on that.

/cc @1ec5

Reply:
By "Radius", I'm referring to the check being done [here.](https://github.com/mapbox/mapbox-navigation-ios/pull/910/files#diff-1a2149692b3d7c58b6a0392120d6011aR617)